### PR TITLE
Refactor PluginManager advanced page handling a bit

### PR DESCRIPTION
**Pull request by olivergondza**
- Redirect back to *advanced* page after saving site url.
- Reduced the scope of `catch(Exception)` block.

### CODE REVIEW

File: core/src/main/java/hudson/PluginManager.java [hunk 1]
```diff
@@ -1604,9 +1609,7 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
                     element("dependencies", dependencies);
             new UpdateSite(UpdateCenter.ID_UPLOAD, null).new Plugin(UpdateCenter.ID_UPLOAD, cfg).deploy(true);
             return new HttpRedirect("../updateCenter");
-        } catch (IOException e) {
-            throw e;
-        } catch (Exception e) {// grrr. fileItem.write throws this
+        } catch (FileUploadException e) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/PluginManager.java]: It may impact RuntimeExceptions if they somehow happen within the code

Comment (daniel-beck) [inline core/src/main/java/hudson/PluginManager.java]: They probably shouldn't have been caught in this catch-all net. As this is just a Stapler call I wouldn't worry too much about it.

Comment (oleg-nenashev) [inline core/src/main/java/hudson/PluginManager.java]: > As this is just a Stapler call I wouldn't worry too much about it.

OH oh, OK

File: core/src/main/java/hudson/PluginManager.java [hunk 2]
```diff
@@ -1524,10 +1525,9 @@ public HttpResponse doSiteConfigure(@QueryParameter String site) throws IOExcept
         }
         sites.add(new UpdateSite(UpdateCenter.ID_DEFAULT, site));
 
-        return HttpResponses.redirectToContextRoot();
+        return new HttpRedirect("advanced");
```
Comment (daniel-beck) [inline core/src/main/java/hudson/PluginManager.java]: I always hated this :+1:

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/java/hudson/PluginManager.java (hunk 2 above)
Comment (oleg-nenashev) [inline core/src/main/java/hudson/PluginManager.java]: +1

**Review by oleg-nenashev — APPROVED** [review]
🚢 🇮🇹

**Pull request merged**